logger/logger.go

```diff
@@ -33,13 +36,29 @@ func (t *telegrafLog) Write(b []byte) (n int, err error) {
 	return t.writer.Write(line)
 }
 
+func (t *telegrafLog) Close() error {
+	closer, isCloser := t.writer.(io.Closer)
+	if !isCloser {
+		return errors.New("the underlying writer cannot be closed")
+	}
+	return closer.Close()
+}
+
 // SetupLogging configures the logging output.
-//   debug   will set the log level to DEBUG
-//   quiet   will set the log level to ERROR
-//   logfile will direct the logging output to a file. Empty string is
-//           interpreted as stderr. If there is an error opening the file the
-//           logger will fallback to stderr.
-func SetupLogging(debug, quiet bool, logfile string) {
+//   debug                  will set the log level to DEBUG
+//   quiet                  will set the log level to ERROR
+//   logfile                will direct the logging output to a file. Empty string is
+//                          interpreted as stderr. If there is an error opening the file the
+//                          logger will fallback to stderr.
+//   logRotationInterval    will rotate when current file at the specified time interval.
+//   logRotationMaxSize     will rotate when current file size exceeds this parameter.
+//   logRotationMaxArchives maximum rotated files to keep (older ones will be deleted)
+func SetupLogging(debug, quiet bool, logfile string, logRotationInterval internal.Duration, logRotationMaxSize internal.Size, logRotationMaxArchives int) {
+	setupLoggingAndReturnWriter(debug, quiet, logfile, logRotationInterval, logRotationMaxSize, logRotationMaxArchives)
+}
+
+func setupLoggingAndReturnWriter(debug, quiet bool, logfile string, logRotationInterval internal.Duration, logRotationMaxSize internal.Size,
```

Comment:
I needed to get the underlying writer so I could close it at the end (otherwise I couldn't clean up the temp test folder/files). 
https://github.com/influxdata/telegraf/blob/cd921626b42eec0beea2ec099197f6ec6515ee15/logger/logger_test.go#L94
SetupLogging doesn't return the telegrafWriter, which is a private struct. So I created this private method that returns the private structure for testability purposes

Reply:
Maybe just change the name of the inner function to `newLogWriter`?